File: eachit.jl
# Test on Julia 0.3

macro each_it(arr, body)
    quote
        for it in $arr
            $body
        end
    end
end

my_array = [1, 2, 3, 4]

@each_it my_array begin
    println("it: $it")
    one_more = it + 1
    println("one_more: $one_more")
end

File: eachit.js
// Compile with:
// $ sjs -o output.js swap.js
// Run with:
// $ node --harmony output.js
// (requires appending "use strict";)

macro eachIt {
    rule { ($x) {$y} } => {
        for (var i=0; i<$x.length; i++) {
            var it = $x[i];
            $y;
        }
    }
}

var myArray = [1, 2, 3];
eachIt(myArray) {
    console.log(it);
}
